snakemake/persistence.py

```diff
@@ -16,6 +16,8 @@
 from itertools import filterfalse, count
 from pathlib import Path
 
+from humanfriendly import format_size
```

Comment:
You have to move this import into the method where you use it. The reason is that Snakemake pickles a part of its code (the class for the snakemake object in scripts) into scripts when executing them. And this fails in the container case whenever there is an import to an external dependency. I am not happy about this and for long already wanted to change that behavior by replacing the pickling with a code generator. But I haven't had the time yet.

Reply:
Ohhhh I see! Okay, I'll do that now

Reply:
That did the trick